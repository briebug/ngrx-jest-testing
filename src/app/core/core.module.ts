import { CommonModule } from '@angular/common';
import {
  ModuleWithProviders,
  NgModule,
  Optional,
  SkipSelf
} from '@angular/core';

import { CustomersService } from '@core/services/customer.service';

@NgModule({
  imports: [CommonModule],
  providers: [CustomersService]
})
export class CoreModule {
  static forRoot(): ModuleWithProviders {
    return {
      ngModule: CoreModule
    };
  }

  constructor(
    @Optional()
    @SkipSelf()
    parentModule: CoreModule
  ) {
    if (parentModule) {
      throw new Error(
        'CoreModule is already loaded. Import it in the AppModule only'
      );
    }
  }
}
