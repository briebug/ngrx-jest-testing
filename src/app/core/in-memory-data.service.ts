import { Customer } from '@state/customer/customer.model';

export class InMemoryDataService {
  createDb() {
    const customers = [
      { id: 1, name: 'Bob Newman' } as Customer,
      { id: 2, name: 'Homer Simpson' } as Customer,
      { id: 3, name: 'Tom Slick' } as Customer,
      { id: 4, name: 'Jane Doe' } as Customer
    ];

    return { customers };
  }
}
