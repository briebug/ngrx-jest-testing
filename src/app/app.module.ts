import { BrowserModule } from '@angular/platform-browser';
import { NgModule } from '@angular/core';

import { AppRoutingModule } from './app.routing.module';
import { AppComponent } from './app.component';
import { StateModule } from './state/state.module';
import { CoreModule } from './core/core.module';
import { HttpClientModule } from '@angular/common/http';
import { InMemoryWebApiModule } from 'angular-in-memory-web-api';
import { InMemoryDataService } from '@core/in-memory-data.service';

@NgModule({
  declarations: [AppComponent],
  imports: [
    BrowserModule,
    AppRoutingModule,
    HttpClientModule,
    StateModule.forRoot(),
    CoreModule.forRoot(),
    InMemoryWebApiModule.forRoot(InMemoryDataService, { delay: 600 })
  ],
  bootstrap: [AppComponent]
})
export class AppModule { }
