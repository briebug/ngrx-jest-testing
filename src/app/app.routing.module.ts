import { NgModule } from '@angular/core';
import { RouterModule, Routes } from '@angular/router';

import { CustomersRoutingModule } from './customers/customers.routing.module';

const routes: Routes = [
  {
    path: '',
    redirectTo: '/customers',
    pathMatch: 'full'
  },
  {
    path: 'customers',
    loadChildren: 'app/customers/customers.routing.module#CustomersRoutingModule'
  }
];

@NgModule({
  imports: [RouterModule.forRoot(routes)],
  exports: [RouterModule]
})
export class AppRoutingModule { }
