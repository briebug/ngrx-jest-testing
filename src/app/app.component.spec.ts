import { TestBed, ComponentFixture, async } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { configureTests, ConfigureFn } from '../test-config.helper'
import { AppComponent } from './app.component';

describe('AppComponent', () => {
  let fixture: ComponentFixture<AppComponent>,
    app: AppComponent;

  beforeEach(
    async(() => {
      const configure: ConfigureFn = testBed => {
        testBed.configureTestingModule({
          declarations: [AppComponent],
          imports: [RouterTestingModule]
        });
      };

      configureTests(configure).then(testBed => {
        fixture = testBed.createComponent(AppComponent);
        app = fixture.componentInstance;
        fixture.detectChanges();
      });
    })
  );

  it('should create the app', async(() => {
    expect(app).toBeTruthy();
  }));

  it('should render html', async(() => {
    const compiled = fixture.debugElement.nativeElement;
    expect(compiled).toMatchSnapshot();
  }));
});
