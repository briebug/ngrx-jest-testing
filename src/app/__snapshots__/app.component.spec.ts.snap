// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should render html 1`] = `
<div
  id="root1"
  ng-version="5.2.9"
/>
`;
